### PR TITLE
ESP32 NimBLE: While starting new connectable advertisement procedure stop any active advertisement

### LINKED ISSUE #3401: [ES32] stops advertising on unsubscribe when built with NimBLE
repro: `chip-tool echo ble 3840 12345678` on a Mac

Then terminate chip-tool to unsubscribe.

```
I (42415) chip[DL]: CHIPoBLE unsubscribe received
I (42425) chip[DL]: BLE GAP connection terminated (con 32387
I (42965) chip[BLE]: Releasing end point's BLE connection back to application.
E (42965) chip[DL]: Long dispatch time: 543 ms
I (42965) chip[DL]: Setup PIN discriminator not found; using default: f00
I (42975) chip[DL]: Configuring CHIPoBLE advertising (interval 200 ms, connectable, device name CHIP-3840)
E (42985) chip[DL]: ble_gap_adv_start() failed: Error 2 (0x00000002)
E (42985) chip[DL]: Configure Adv Data failed: Error 2 (0x00000002)
E (42995) chip[DL]: Disabling CHIPoBLE service due to error: Error 2 (0x00000002)
```

Device stops advertising here, instead of restarting advertising with 200ms interval.

@prasad-alatkar

### PR DESCRIPTION
**Pull request by prasad-alatkar**
#### Problem
When non-connectable advertisement is in progress and the BLE connection is closed (disconnected), then trying to restart connectable advertising fails as NimBLE host stack returns error code `BLE_HS_EALREADY` i.e. "Operation already in progress".

 #### Summary of Changes
Stop ongoing/active advertisement (connectable/non-connectable) whenever connectable advertisement is attempted through DriveBLEState. Some minor miscellaneous changes. 

 Fixes #3401

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

**Review by rwalker-apple — APPROVED** [review]

File: src/platform/ESP32/nimble/BLEManagerImpl.cpp [hunk 1]
```diff
@@ -1057,6 +1055,21 @@ CHIP_ERROR BLEManagerImpl::StartAdvertising(void)
...
 
 exit:
+    if (err == BLE_HS_EALREADY && connectable)
+    {
+        ChipLogProgress(DeviceLayer,
+                        "Connectable advertising failed because device was already advertising , stop active advertisement");
+        err = ble_gap_adv_stop();
+        if (err != CHIP_NO_ERROR)
+        {
+            ChipLogError(DeviceLayer, "ble_gap_adv_stop() failed: %s", ErrorStr(err));
```
Comment (dhrishi) [inline src/platform/ESP32/nimble/BLEManagerImpl.cpp:1065]: @prasad-alatkar  Do you want to continue to start advertisement even if stopping the advertisement has failed?

**Comment by mspang** [comment]
I tried to look into this myself and got fairly confused. I'm not sure about this fix, either.

Is the non-connectable advertisement actually working right now (are we sending out beacons while connected)? It seems like the connectable advertisement ceases when the connection is established, and resumes automatically on disconnect. And my read was that the non-connectable advertisement probably wasn't working at all.

**Comment by prasad-alatkar** [comment]
> Is the non-connectable advertisement actually working right now (are we sending out beacons while connected)? It seems like the connectable advertisement ceases when the connection is established, and resumes automatically on disconnect. And my read was that the non-connectable advertisement probably wasn't working at all.

@mspang , yes the non-connectable advertisement does work. As also mentioned in problem description of this PR, the connectable advertisement fails to restart (after disconnect) because the non-connectable advertisement was already in progress. So this PR restarts connectable advertisement (after disconnect) by stopping non-connectable advertisement (which was already in progress). Please let me know if you still have any concerns.

File: src/platform/ESP32/nimble/BLEManagerImpl.cpp (hunk 1 above)
Comment (prasad-alatkar) [reply src/platform/ESP32/nimble/BLEManagerImpl.cpp:1065]: Will make appropriate change. Thanks!!

Comment (prasad-alatkar) [reply src/platform/ESP32/nimble/BLEManagerImpl.cpp:1065]: Done!!

File: src/platform/ESP32/nimble/BLEManagerImpl.cpp [hunk 2]
```diff
@@ -1057,6 +1055,24 @@ CHIP_ERROR BLEManagerImpl::StartAdvertising(void)
     }
 
 exit:
+    if (err == BLE_HS_EALREADY && connectable)
+    {
+        ChipLogProgress(DeviceLayer,
+                        "Connectable advertising failed because device was already advertising , stop active advertisement");
```
Comment (BroderickCarlin) [inline src/platform/ESP32/nimble/BLEManagerImpl.cpp:1061]: nit: extra space before ","

**Review by BroderickCarlin — APPROVED** [review]

**Review by mspang — APPROVED** [review]

**Pull request merged**